### ISSUE TITLE
[Test]: Add e2e test of BuildCheck for severity set to Error

### ISSUE THREAD

**Issue by f-alizada** [dotnet/msbuild#10386]
### Summary

Once the PR reenabling the e2e tests is merged: https://github.com/dotnet/msbuild/pull/10307 
Let's add the automated test to cover the scenario:
BuildCheck severity set to Error -> Build of the project actually fails. 
This item is follow up of : https://github.com/dotnet/msbuild/pull/10227

**Comment by YuliiaKovalova** [comment]
This is already covered in scope of [existing tests](https://github.com/dotnet/msbuild/pull/10496/commits/41fbe9cb0eea9d119150600095145027dcb05925), but didn't work due to  https://github.com/dotnet/msbuild/issues/10176
